### PR TITLE
feat: Add auth filter without subdomain validation

### PR DESCRIPTION
**Pull request by thoriqsatriya**

### CODE REVIEW

**Review by medreza — APPROVED** [review]

File: pkg/auth/iam/iam.go [hunk 1]
```diff
@@ -208,6 +208,90 @@ func (filter *Filter) Auth(opts ...FilterOption) restful.FilterFunction {
...
+// The token's claims will be passed in the request.attributes["JWTClaims"] = *iam.JWTClaims{}
+// This filter is expandable through FilterOption parameter
+// Example:
+// iam.AuthWithoutSubdomainValidation(
+//
+//	WithValidUser(),
+//	WithPermission("ADMIN"),
+//
+// )
+func (filter *Filter) AuthWithoutSubdomainValidation(opts ...FilterOption) restful.FilterFunction {
```
Comment (wardab0104) [inline pkg/auth/iam/iam.go:223]: when should we use this function instead of **Auth()**?

Comment (thoriqsatriya) [reply pkg/auth/iam/iam.go:223]: When the endpoints need to be called from page with no subdomain. For example, all endpoints that needed to be called from self-service checkout page

Comment (angelo-chan) [reply pkg/auth/iam/iam.go:223]: I have no concerns about the design, but the implementation is not good with so many duplicates

Comment (thoriqsatriya) [reply pkg/auth/iam/iam.go:144]: Thanks for the suggestion. I've moved it to a function and add parameter to it to remove the duplication

**Review by angelo-chan — APPROVED** [review]

**Pull request merged**